### PR TITLE
Fix bug

### PR DESCRIPTION
**Pull request by G-Ricky**
Add a temporary array in _reorderNFAStateId to save ordered results.

### CODE REVIEW

File: src/regparser.js [hunk 1]
```diff
@@ -297,6 +298,10 @@ RegParser.prototype._reorderNFAStateId = function() {
       queue.push(state.nextStates[i][1]);
     }
   }
+  while(ordered.length) {
```
Comment (hokein) [inline src/regparser.js:301]: nit: a space between `while` and `(`.

**Review by hokein — COMMENTED** [review]
@G-Ricky Awesome!
The code looks mostly good, could you please add a testcase in `test/test.js`?

**Comment by G-Ricky** [comment]
Sorry, I don't know much about node.js.I just debug the browser version.

**Comment by hokein** [comment]
@G-Ricky no worries, I have taken care of it. Thanks for the contribution, the latest sample should pick up the fix.

**Pull request merged**